web/views.py

```
@@ -718,12 +719,8 @@ def course_detail(request, slug):
     # Get the calendar for current month
     cal = calendar.monthcalendar(current_month.year, current_month.month)
 
-    # Get all session dates for this course in current month
-    session_dates = set(
-        session.start_time.date()
-        for session in sessions
-        if session.start_time.year == current_month.year and session.start_time.month == current_month.month
-    )
+    # Get all session dates for this course
+    session_dates = set(session.start_time.date() for session in sessions)
```

Ruff (0.8.2)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)